docs/source/python/install.rst

```diff
@@ -93,3 +100,41 @@ a custom path to the database from Python:
 
    >>> import pyarrow as pa
    >>> pa.set_timezone_db_path("custom_path")
+
+
+.. _python-conda-differences:
+
+Differences between conda-forge packages
+----------------------------------------
+
+PyArrow is packaged on `conda-forge <https://conda-forge.org/>`_ as three
+separate packages, each providing varying levels of functionality. This is in
+contrast to PyPi, where only a single PyArrow package is provided.
+
+The purpose of this split is to minimize the size of the installed package for
+most users (``pyarrow``), provide a smaller, minimal package for specialized use
+cases (``pyarrow-core``), while still providing a complete package for users who
+require it (``pyarrow-all``).
+
+The table below lists the functionality provided by each package and may be
+useful when deciding to use one package over another:
+
++------------+------------------------------+------------------------------+------------------------------+
+| Component  | pyarrow                      | pyarrow-core                 | pyarrow-all                  |
++============+==============================+==============================+==============================+
+| Core       | :fas:`check;sd-text-success` | :fas:`check;sd-text-success` | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Parquet    | :fas:`check;sd-text-success` |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Datasets   | :fas:`check;sd-text-success` |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Acero      | :fas:`check;sd-text-success` |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Substrait  | :fas:`check;sd-text-success` |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Flight     |                              |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Flight SQL |                              |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
+| Gandiva    |                              |                              | :fas:`check;sd-text-success` |
++------------+------------------------------+------------------------------+------------------------------+
```

Comment:
The rows in the table above are based on @raulcd's breakdown on the mailing list but I think we want to be careful about what's listed here. I think the best thing would be for this table to fully align with the submodules exposed in PyArrow since that's what the user is most familiar with. And we might even consider renaming the "Component" column to "Module" and using the literal module names. i.e., `parquet` instead of Parquet so it's clear we're talking about being able to run `import pyarrow.parquet` or not.

This would mean the table isn't complete yet (json, csv, filesystems, orc, more?).

Reply:
Agree with aligning with PyArrow on the names 👍

Reply:
In looking closer, I think it makes sense to align on conda package names too so in https://github.com/apache/arrow/pull/41135/commits/cccc7ff655ae34420214d3b512aafaebc6775942 I updated the table to include the names of the conda package for each component.

I need to look at where the modules I mentioned above (JSON, etc) live because I think that should be clear.

Reply:
> I need to look at where the modules I mentioned above (JSON, etc) live because I think that should be clear.

Right now all the things you mentioned (json, csv, filesystems, orc) are included in the main libarrow.so, and so can only be either enabled or disabled (cannot be installed separately). 
For the filesystems, there is work under way to actually build them as separate libraries, so we will have to update this doc section once that is done. For ORC I am actually wondering if we should open an issue about splitting that in a separate library (although liborc is not that big of a dependency, so it might not be super important)

Reply:
Thanks both of you. The latest draft now tries to make it clear which Python modules each package provides. It'd be good to get reviews on the overall style and also the correctnes/completeness.